### PR TITLE
include custom scalars in merge_subgraphs

### PR DESCRIPTION
**Pull request by lennyburdette**
---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by pubmodmatt — APPROVED** [review]

File: apollo-federation/src/merge.rs [hunk 1]
```diff
@@ -591,6 +598,25 @@ impl Merger {
...
+        &self,
+        types: &mut IndexMap<Name, ExtendedType>,
+        subgraph_name: Name,
+        scalar_name: NamedType,
+        ty: &Node<ScalarType>,
+    ) {
+        let existing_type = types
+            .entry(scalar_name.clone())
+            .or_insert(copy_scalar_type(scalar_name, ty));
+        if let ExtendedType::Scalar(e) = existing_type {
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply apollo-federation/src/merge.rs:612]: ah yeah good catch, this was me being lazy when i copy-pasted the enum version of this function

File: apollo-federation/src/merge.rs [hunk 2]
```diff
@@ -591,6 +598,25 @@ impl Merger {
...
+            let join_type_directives =
+                join_type_applied_directive(subgraph_name.clone(), iter::empty(), false);
+            e.make_mut().directives.extend(join_type_directives);
+        } else {
+            // conflict?
```
Comment (nicholascioli) [inline apollo-federation/src/merge.rs:617]: Should we change the signature of this method to return a `Result` and have it bubble up? Might help us catch issues early on if we encounter areas where this conflict happens.

**Review by nicholascioli — COMMENTED** [review]
Just a small comment on returning errors. Looks great otherwise!

File: apollo-federation/src/merge.rs (hunk 2 above)
Comment (lennyburdette) [reply apollo-federation/src/merge.rs:617]: this was a copy-paste from a similar method. i'll add a subtask to the merge ticket to return to the conflict handling

**Pull request merged**